### ISSUE TITLE
Upload progress + PHP 7.2

### ISSUE THREAD

**Issue by fesarlis** [roundcube/roundcubemail#6177]
Hello.
What is the current way to enable upload progress in RC+PHP 7.2; uploadprogress pecl extension does not compile anymore. Found one that claims it works (https://github.com/Jan-E/uploadprogress), it compiles fine but RC does not attach files, it produces an error.
RC = 1.3.2

Thanks

**Comment by alecpl** [comment]
What error? Roundcube supports also apc.rfc1867 (with apc.rfc1867_name) or session.upload_progress.enabled (with session.upload_progress.name), but I didn't test these recently.

**Comment by fesarlis** [comment]
Did not try apc.rfc1867 since apc is now obsolete. Error appears when trying to compile pecl extension under PHP 7.2. As I said there is one (link supplied above) that compiles but doesn't work. Tried to togge session.upload_progress.name with no luck (I've never had to use it with PHP 5.6 though). I paste part of the error when trying to  compile uploadprogress in PHP 7.2

` cc -I. -I/tmp/pear/temp/uploadprogress -DPHP_ATOM_INC -I/tmp/pear/temp/pear-build-rootddPNtt/uploadprogress-1.0.3.1/include -I/tmp/pear/temp/pear-build-rootddPNtt/uploadprogress-1.0.3.1/main -I/tmp/pear/temp/uploadprogress -I/opt/servers/php/include/php -I/opt/servers/php/include/php/main -I/opt/servers/php/include/php/TSRM -I/opt/servers/php/include/php/Zend -I/opt/servers/php/include/php/ext -I/opt/servers/php/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pear/temp/uploadprogress/uploadprogress.c  -fPIC -DPIC -o .libs/uploadprogress.o
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function ‘uploadprogress_php_rfc1867_file’:
/tmp/pear/temp/uploadprogress/uploadprogress.c:160:31: error: ‘ENFORCE_SAFE_MODE’ undeclared (first use in this function)
                 int options = ENFORCE_SAFE_MODE;
                               ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:160:31: note: each undeclared identifier is reported only once for each function it appears in
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function ‘uploadprogress_file_php_get_info’:
/tmp/pear/temp/uploadprogress/uploadprogress.c:420:57: error: macro "add_assoc_string" passed 4 arguments, but takes just 3
                 add_assoc_string( return_value, k, v, 1 );
                                                         ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:420:17: error: ‘add_assoc_string’ undeclared (first use in this function)
                 add_assoc_string( return_value, k, v, 1 );
                 ^
/tmp/pear/temp/uploadprogress/uploadprogress.c: In function ‘uploadprogress_file_php_get_contents’:
/tmp/pear/temp/uploadprogress/uploadprogress.c:437:19: error: ‘ENFORCE_SAFE_MODE’ undeclared (first use in this function)
     int options = ENFORCE_SAFE_MODE;
                   ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:458:71: error: macro "php_stream_copy_to_mem" passed 4 arguments, but takes just 3
         if ((len = php_stream_copy_to_mem(stream, &contents, maxlen, 0)) > 0) {
                                                                       ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:458:20: error: ‘php_stream_copy_to_mem’ undeclared (first use in this function)
         if ((len = php_stream_copy_to_mem(stream, &contents, maxlen, 0)) > 0) {
                    ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:466:44: error: macro "RETVAL_STRINGL" passed 3 arguments, but takes just 2
             RETVAL_STRINGL(contents, len, 0);
                                            ^
/tmp/pear/temp/uploadprogress/uploadprogress.c:466:13: error: ‘RETVAL_STRINGL’ undeclared (first use in this function)
             RETVAL_STRINGL(contents, len, 0);
             ^
make: *** [uploadprogress.lo] Error 1
ERROR: `make' failed`

Therefore it is my understanding that RC must now find a different way to support upload progress.

**Comment by alecpl** [comment]
I think apc.rfc1867 should work with APCu extension.

**Comment by alecpl** [comment]
Session handler does not work for me. It also looks that APCu 5.x does not support apc.rfc1867. I found some hints on https://www.drupal.org/project/drupal/issues/2718253. There's php-uploadprogress package on ubuntu that is supposed to work with PHP 7.2, but does not work for me for unknown reason.

**Comment by alecpl** [comment]
Proposed solution in #6583. @thomascube what do you think? how about merging it for 1.4?

**Comment by thomascube** [comment]
@alecpl Looks good. The new approach includes `apc.rfc1867` and thus should also work on PHP 5.x versions, right? 1.4 RC packages are almost ready but we could still include this.

**Comment by alecpl** [comment]
To clarify, by HTML5-only I mean upload progress is javascript-client-side thing. Also much better for performance.

**Comment by itcrowd72** [comment]
I use debian repo https://packages.sury.org/php/ it has php 7.1, 7.2, 7.3 and package php-uploadprogress for all these versions

**Comment by alecpl** [comment]
HTML5 solution merged into master.
